augur/dates/ambiguous_date.py

```diff
@@ -6,6 +6,10 @@
 from .errors import InvalidDate
 
 
+# 'X' followed by a specific digit does not make sense.
+RE_INVALID_AMBIGUITY = re.compile(r'.*X[0-9]+.*')
```

Comment:
_non-blocking_

The surrounding `.*` are only required here because the `re.match()` method is used below. If we switched to `re.search()`, then the pattern can simplify to `X[0-9]+`.